### PR TITLE
DB-3177: PostGrid

### PR DESCRIPTION
**Pull request by enZane**
## What changes were made?
- Added to the components the Grid component with similar behavior to the drupal starter
    - There is a HOC called withGrid
    - Also there is the PostGridItem component to render a Post
- Added  to the libs, the Post file that has the function getLatestPosts
- In the pages/index component, the grid component and the getLatestPosts are used to display the PostGrid

## Where were the changes made?
- next-wordpress-starter

## How have the changes been tested?
Tested in development and builded

## Additional information

Don't forget to [add a changeset](https://github.com/pantheon-systems/decoupled-kit-js#generating-a-changeset) if needed!

### CODE REVIEW

File: starters/next-wordpress-starter/components/grid.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    <Link passHref href={post.uri}>
+      <a>
+        <div className="flex flex-col rounded-lg shadow-lg overflow-hidden cursor-pointer border-2 h-full hover:border-indigo-500">
+          <div className="flex-shrink-0 relative h-40">
+            {imgSrc !== "" ? (
+              <Image
+                src={imgSrc}
+                layout="fill"
+                objectFit="cover"
+                alt={post.title}
```
Comment (CobyPear) [inline starters/next-wordpress-starter/components/grid.js:40]: Since there is `altText` available on the `featuredImage` node, let's use it here.

**Review by CobyPear — CHANGES_REQUESTED** [review]
One small change requested. Looks good tho!!

File: starters/next-wordpress-starter/next.config.js [hunk 2]
```diff
@@ -12,6 +12,9 @@ const nextConfig = {
   env: {
     backendUrl: process.env.WPGRAPHQL_URL,
   },
+  images: {
+    domains: [process.env.IMAGE_DOMAIN],
```
Comment (CobyPear) [inline starters/next-wordpress-starter/next.config.js:16]: Oh one more thing, can you add this env var to the `.env.example`?

Comment (enZane) [reply starters/next-wordpress-starter/next.config.js:16]: Sure

**Review by CobyPear — APPROVED** [review]
Nice job, @enZane. LGTM!

**Pull request merged**